File: invert-binary-tree/eunhwa99.java
class Solution {

  public TreeNode invertTree(TreeNode root) {
    if (root == null) {
      return null;
    }

    TreeNode tmp = root.left;
    root.left = root.right;
    root.right = tmp;

    invertTree(root.right);
    invertTree(root.left);

    return root;
  }

}

File: kth-smallest-element-in-a-bst/eunhwa99.kt
class Solution {
    // TC : O(H+K)
    // SC: O(H)
    private var K: Int = 0
    fun kthSmallest(root: TreeNode?, k: Int): Int {
        K = k
        return traverse(root)!!.`val`
    }

    fun traverse(cur: TreeNode?): TreeNode?{
        if(cur == null) return cur

        val result = traverse(cur.left)
        K--
        if(K == 0) return cur
        return result ?: traverse(cur.right)
    }
}
